tests/test_config.py

```diff
@@ -155,6 +156,97 @@ def test_use_https():
         assert config.local_install
         assert not config.is_secure
 
+@patch('helpers.aws_validation.AWSValidation.validate_credentials',
+       new=MockAWSValidation.validate_credentials)
+def test_aws_validation():
+    config = read_config()
+
+    assert not config._Config__dict['use_aws']
+    assert not config._Config__dict['aws_credentials_valid']
+
+    with patch('helpers.cli.CLI.colored_input') as mock_colored_input:
+        mock_colored_input.side_effect = iter([CHOICE_NO])
+        assert not config._Config__dict['use_aws']
+        assert not config._Config__dict['aws_credentials_valid']
+
+    with patch('helpers.cli.CLI.colored_input') as mock_colored_input:
+        try:
+            mock_colored_input.side_effect = iter(
+                [CHOICE_YES, '', '', '', CHOICE_YES, '', '', '', '', '', '']
+            )
```

Comment:
We may want to restrict the try/except scope to what it tries to catch.

```python3
        mock_colored_input.side_effect = iter(
            [CHOICE_YES, '', '', '', CHOICE_YES, '', '', '', '', '', '']
        )
        try:
            config._Config__questions_aws()
        except SystemExit:
            pass
```

Ok, ok! To be fair, I have to admit I used to do the same as your code and @jnm brought be back on the correct path :-D